### DISCUSSION TITLE
Is there a way to clear temporary rules after site/tab close or timeout?

### DISCUSSION THREAD

**Discussion by tomasz1986** [uBlockOrigin/uBlock-issues#2429] [Q&A] [answered]
Currently, when using buttons in the pop-up, e.g. to unblock images/fonts/JS, new rules are added to the temporary rules list, and they stay there until the browser is restarted. However, I would very much prefer for them to be automatically cleared after closing the specific site/tab, or if that's not feasible, maybe after a user-set timeout. Is something like this even possible?

The first reason is that I often need to make manual edits to the rules list, and when doing so, I always need to be very careful not to add those temporary rules to the permanent rules list together with my custom additions and modifications (which happens upon pressing the "Commit" button). The second reason is that I tend to keep the browser open for days, so those temporary rules become kind of semi-permanent in my use case (unless I manually disable them after being finished with the specific site which I usually forget to do 🙁).

**Comment by ghost** [comment]
Click on "Revert".

**Reply by tomasz1986** [reply]
Yeah, that's an option but the problem is that after I've added my own modifications, this will revert them too. I would need to always remember to press that before modifying anything manually, which knowing myself isn't going to happen, as I will probably just keep forgetting to do so.

**Reply by uBlock-user** [reply]
He's asking for something that's automated after a certain period of time.

**Reply by tomasz1986** [reply]
An automated solution would be ideal, but I think I've gone through all the documentation and I haven't been able to find any setting that would make this possible. The only similarly sounding option seems to be https://github.com/gorhill/uBlock/wiki/Advanced-settings#strictblockingbypassduration, but it's used for something completely different.

**Reply by Yuki2718** [reply]
What about filtering by temporal state in My rules (not possible currently)? Then you can remove all the temporal rules easily at once. I felt this would be convinient several times.

**Comment by uBlock-user** [comment, marked as answer]
Put name of the site for which you want to remove the "temp" rules in the rules filter, click revert.

**Reply by u-RraaLL** [reply]
Collapsing the rules should prove useful too.

**Reply by tomasz1986** [reply]
Thank you for a very useful tip, actually. I wasn't aware you could do that.

**Reply by uBlock-user** [reply]
> Collapsing the rules

How do you do that ?

**Reply by u-RraaLL** [reply]
> > Collapsing the rules
> 
> How do you do that ?

The double `^` icon to the right of the search and sort.

**Reply by uBlock-user** [reply]
Right, that one. I thought you were suggesting something else.

**Reply by Yuki2718** [reply]
Didn't know, thanks for the tip! To me the problem was solved.
